src/client/Sidebar/Item.jsx

```diff
@@ -8,6 +14,52 @@ export const Item = ({
 }) => {
   const [isHovered, setIsHovered] = useState(false);
 
+  const [showModal, setShowModal] = useState(false);
+
+  //Function to open the modal
+  const handleModalOpen = useCallback(() => {
+    setShowModal(true);
+  }, []);
+
+  //Function to close the modal
+  const handleModalClose = useCallback(() => {
+    setShowModal(false);
+  }, []);
+
+  //function to handle confirmation on modal
+  const handleConfirmModal = useCallback(() => {
+    setShowModal(false);
+    handleDeleteClick();
+  }, [handleDeleteClick]);
+
+  //function to close Modal if anywhere else on the screen is clicked
+  const outsideClickHandler = useCallback((ref) => {
+    useEffect(() => {
+      //check if clicked outside
+      function handleClickOutside(event) {
```

Comment:
Interesting... I was under the impression that react-bootstrap handled this interaction of clicking outside the modal to close it.